Fix smart pointers errors with CFRunLoopGetMain()/CFRunLoopGetCurrent()

fred-wang opened the pull request:
#### 1af87bb4a5ef066a6347dfb7f51e7ccd8b8284ab
<pre>
Fix smart pointers errors with CFRunLoopGetMain()/CFRunLoopGetCurrent()
<a href="https://bugs.webkit.org/show_bug.cgi?id=298865">https://bugs.webkit.org/show_bug.cgi?id=298865</a>

Reviewed by Chris Dumez.

Ownership follows the The Get Rule, so they need a RetainPtr:
<a href="https://developer.apple.com/documentation/corefoundation/cfrunloopgetcurrent()?language=objc">https://developer.apple.com/documentation/corefoundation/cfrunloopgetcurrent()?language=objc</a>
<a href="https://developer.apple.com/documentation/corefoundation/cfrunloopgetmain()?language=objc">https://developer.apple.com/documentation/corefoundation/cfrunloopgetmain()?language=objc</a>

* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidConnection.mm:
(WebKit::HidConnection::initialize):
(WebKit::HidConnection::terminate):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidService.mm:
(WebKit::HidService::~HidService):
(WebKit::HidService::platformStartDiscovery):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::computeHasVisualSearchResults):
* Source/WebKit/WebProcess/WebPage/WebDisplayRefreshMonitor.cpp:
(WebKit::WebDisplayRefreshMonitor::startNotificationMechanism):
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(WebKit::WebPushToolVerb::done):

Canonical link: <a href="https://commits.webkit.org/300027@main">https://commits.webkit.org/300027@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/839fe42630ea8c38dc09b0869651679bd8b3cd45

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121082 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40778 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31436 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127501 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73163 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41480 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49357 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91972 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61186 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124034 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33122 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108540 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72656 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32146 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71092 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/102628 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26822 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130354 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48009 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36486 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100581 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48377 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104707 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100483 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25473 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45887 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23940 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44703 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47867 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53580 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47338 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50685 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49022 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2645e822b543befc21226e9dbd8c59c32cce33d3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50795

fred-wang committed:
Fix smart pointers errors with CFRunLoopGetMain()/CFRunLoopGetCurrent()
https://bugs.webkit.org/show_bug.cgi?id=298865

Reviewed by Chris Dumez.

Ownership follows the The Get Rule, so they need a RetainPtr:
https://developer.apple.com/documentation/corefoundation/cfrunloopgetcurrent()?language=objc
https://developer.apple.com/documentation/corefoundation/cfrunloopgetmain()?language=objc

* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidConnection.mm:
(WebKit::HidConnection::initialize):
(WebKit::HidConnection::terminate):
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/HidService.mm:
(WebKit::HidService::~HidService):
(WebKit::HidService::platformStartDiscovery):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::computeHasVisualSearchResults):
* Source/WebKit/WebProcess/WebPage/WebDisplayRefreshMonitor.cpp:
(WebKit::WebDisplayRefreshMonitor::startNotificationMechanism):
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(WebKit::WebPushToolVerb::done):

Canonical link: https://commits.webkit.org/300027@main
